build.rs

```diff
@@ -1,22 +1,62 @@
 use std::env;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
+
+fn path_exists(path: &str) -> bool {
+    Path::new(path).exists()
+}
+
+fn find_julia() -> Option<String> {
```

Comment:
The behaviour of this function is different depending on what branch it takes. If JL_PATH is set, it will just assume everything can be found through it. Otherwise it explicitly checks if julia.h exists in /usr/include/julia, failing that it performs the same explicit check for directories in either CPATH or C_INCLUDE_PATH, though it will only check one of the two. If it can't find julia.h, it panics at the end. While I do think it's good to check if /usr/include/julia/julia.h exists if JL_PATH is not set, I'd rather not have the additional (undocumented) checks for CPATH and C_INCLUDE_PATH and panic at the end.

Reply:
What do you wish to do if the `/usr/include/julia/julia.h` check fails?

Reply:
There's also the [pkg-config][1] crate, but I was not able to convince it to find Julia.

[1]: https://crates.io/crates/pkg-config

Reply:
I think it should just be accepted assuming "the user either ensures everything can be found through a default path, finds out they need to use JL_PATH, or they can open an issue if their use case is a hard requirement." My experience with installing Julia on Ubuntu is that the headers are not available, and I couldn't find a julia-dev package either, so I haven't tested it. If things are available the same way as they are in the official archives, your approach should work.

Reply:
What do you think about this?
```rust
fn find_julia() -> Option<String> {
    if let Ok(path) = env::var("JL_PATH") {
        return Some(path);
    }

    if path_exists("/usr/include/julia/julia.h") {
        return Some("/usr".to_string());
    }

    // This message is not displayed to the user unless the compile fails.
    eprintln!("You can specify the Julia installation path with the JL_PATH environment variable.");

    None
}
```
As for Ubuntu, I believe the package with the headers is called [`libjulia-dev`](https://packages.debian.org/unstable/libjulia-dev).

Reply:
Looks great! I'll check if I can find libjulia-dev on Ubuntu 19.10 when I get home (I think I was always looking for julia-dev instead). In any case, it's not available on 18.04 as far as I'm aware (but neither is julia itself)